### PR TITLE
chore: remove slim-build from k8s ci

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

### CODE REVIEW

File: .github/workflows/k8s_e2e.yml [hunk 1]
```diff
@@ -165,7 +165,7 @@ jobs:
           KUBERNETES_VERSION: ${{ matrix.kubernetes_version.version }}
           MINIKUBE_VERSION: ${{ matrix.minikube_version }}
           CONTAINER_RUNTIME: ${{ matrix.container_runtime }}
-      - run: make slim-builds
+          CARGO_INCREMENTAL: 0
```
Comment (jszwedko) [inline .github/workflows/k8s_e2e.yml:168]: I think this should be moved up to the job level otherwise it only applies to the minikube setup step.

Comment (spencergilbert) [reply .github/workflows/k8s_e2e.yml]: Moved it to the right env section...

**Review by jszwedko — APPROVED** [review]
Thanks for catching this!

**Comment by jszwedko** [comment]
Regression analysis is a false positive since we just merged in #11587

**Pull request merged**